awscrt/http.py

```diff
@@ -127,14 +150,18 @@ def new(cls,
             if not bootstrap:
                 bootstrap = ClientBootstrap.get_or_create_static_default()
 
-            connection = cls()
+            connection = expected_cls()
```

Comment:
we could always delay instantiating `connection` with the proper class type until we know the actual http_version, later in on_connection_setup

Like, we could go back to just having 1 HttpClientConnection.new() function, and you may end up with an Http2ClientConnection or HttpClientConnection instance, depending